core/windows_and_tabs/window_management.talon

```diff
@@ -17,5 +17,43 @@ snap last [screen]: user.move_window_previous_screen()
 snap screen <number>: user.move_window_to_screen(number)
 snap <user.running_applications> <user.window_snap_position>:
     user.snap_app(running_applications, window_snap_position)
+snap left <user.running_applications> <user.running_applications>:
+    user.switcher_focus(running_applications_1)
+    user.snap_window_to_position("left")
+    user.switcher_focus(running_applications_2)
+    user.snap_window_to_position("right")
+snap left <user.running_applications> <user.running_applications> <user.running_applications>:
+    user.switcher_focus(running_applications_1)
+    user.snap_window_to_position("left third")
+    user.switcher_focus(running_applications_2)
+    user.snap_window_to_position("center third")
+    user.switcher_focus(running_applications_3)
+    user.snap_window_to_position("right third")
+snap right <user.running_applications> <user.running_applications>:
+    user.switcher_focus(running_applications_2)
+    user.snap_window_to_position("left")
+    user.switcher_focus(running_applications_1)
+    user.snap_window_to_position("right")
+snap right <user.running_applications> <user.running_applications> <user.running_applications>:
+    user.switcher_focus(running_applications_3)
+    user.snap_window_to_position("left third")
+    user.switcher_focus(running_applications_2)
+    user.snap_window_to_position("center third")
+    user.switcher_focus(running_applications_1)
+    user.snap_window_to_position("right third")
+snap clock <user.running_applications> <user.running_applications> <user.running_applications>:
+    user.switcher_focus(running_applications_2)
+    user.snap_window_to_position("top right")
+    user.switcher_focus(running_applications_3)
+    user.snap_window_to_position("bottom right")
+    user.switcher_focus(running_applications_1)
+    user.snap_window_to_position("left")
+snap counterclock <user.running_applications> <user.running_applications> <user.running_applications>:
+    user.switcher_focus(running_applications_2)
+    user.snap_window_to_position("top left")
+    user.switcher_focus(running_applications_3)
+    user.snap_window_to_position("bottom left")
+    user.switcher_focus(running_applications_1)
+    user.snap_window_to_position("right")
```

Comment:
As you're getting into many commands, this feels like a good candidate for a list that would index into a data structure mapping the name of the arrangement to the list of window positions (a bit like to how `_snap_positions` works today). This would reduce the duplication and make it much easier to add new arrangements.

Then you would just have one command definition:

```
snap {user.window_arrangement} <user.running_applications>+: user.snap_arrangement(window_arrangement, running_applications_list)
```

Reply:
I like this idea. I'm still pretty uncomfortable with building these dynamic workflows from scratch. Please feel free to change this if it's obvious hope to do so-I'll give it a shot when I have some time.

Reply:
Sure, I'd be happy to do this or walk through it if you want to pair. Either way is fine

Reply:
Thank you for the offer. Pairing sounds nice if I can find time. Let me think about it a bit.

Reply:
Given that it's a bit more complex than I realized (because we have to indicate which window will be the active one after arrangement, to your point above), it might be best if I prototype it quickly first, just to make sure it's as straightforward as I think it is. But I'm still happy to pair on it afterwards (or anything else, for that matter)

Reply:
Sounds good. Please go ahead-looking forward to seeing what you put together.